examples/gaussian_process/svgp.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+'''
+Sparse variational Gaussian process with normal approximation for posterior.
+
+For the formulation you can refer to, e.g., Section 2.1 of the following paper:
+
+Salimbeni and Deisenroth 2017, Doubly Stochastic Variational Inference for Deep
+Gaussian Processes.
+
+Results (RMSE, NLL):
+ N_Z   Boston    Protein
+----- --------- ---------
+ 100  2.73,2.50
+ 500            4.08,2.86
+'''
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+from six.moves import range, zip
+
+import numpy as np
+import tensorflow as tf
+import zhusuan as zs
+from zhusuan.utils import log_mean_exp
+from examples import conf
+from examples.utils import dataset
+import os
+import argparse
+from utils import gp_conditional, RBFKernel
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-n_z', default=100, type=int)
+parser.add_argument('-n_covariates', default=1, type=int)
+parser.add_argument('-n_particles', default=20, type=int)
+parser.add_argument('-n_particles_test', default=100, type=int)
+parser.add_argument('-n_batch', default=5000, type=int)
+parser.add_argument('-n_epoch', default=5000, type=int)
+parser.add_argument('-dtype', default='float32', type=str,
+                    choices=['float32', 'float64'])
+parser.add_argument('-dataset', default='boston_housing', type=str,
+                    choices=['boston_housing', 'protein_data'])
+
+
+class SVGP:
+
+    def __init__(self, hps, n_train, std_y_train):
+        self._hps = hps
+        self._kernel = RBFKernel(hps.n_covariates)
+        self._x_ph = x_ph = tf.placeholder(
+            hps.dtype, [None, hps.n_covariates], 'x')
+        self._y_ph = y_ph = tf.placeholder(
+            hps.dtype, [None], 'y')
+        self._z_pos = tf.get_variable(
+            'z/pos', [hps.n_z, hps.n_covariates], hps.dtype,
+            initializer=tf.random_uniform_initializer(-1, 1))
+        self._n_particles_ph = n_particles_ph = tf.placeholder(
+            tf.int32, [], 'n_particles')
+        n_batch = tf.cast(tf.shape(x_ph)[0], hps.dtype)
```

Comment:
`batch_size` or `n` instead of `n_batch`

Reply:
What about the args? I.e. when should we use `n_*` (e.g. in n_particles), and when `*_size` (e.g. batch)?

Reply:
They are different. There are indeed n particles. But there are not n batches. I used to use n_x or n_z, but now prefer x_dim/z_dim.